### PR TITLE
Style improvement

### PR DESCRIPTION
**Pull request by olavim**
Proposal for new UI. Some functionality is broken (theme selection) and new bugs have been introduced. Never mind that. Please discuss the visual look. Join/create room menu is not in the scope of this discussion, though.

This is not meant to be a finished look. Some stuff like text input styles haven't been touched at all. I'm taking the UI to a certain direction design-wise and want critique.

### CODE REVIEW

**Comment by olavim** [comment]
Apparently heroku doesn't autodeploy draft pull requests. So you need to pull and test locally 😞

**Comment by GourmetHunter** [comment]
I checkout it out, some of the changes are very good, such as how the chat messages look, and the whole right side of the ui (cards). 

However I personally like the colored navigation bar we had before, The bar it self isn't necessarily important, I just think the look could use more color.

I also think the turn timer could be on the right side of the screen together with other "game elements" of the chat.

**Comment by fergusq** [comment]
Currently chat bugs when messages go over the screen. Other thing I noticed is that if user names are changed (anonymized or pseudonymized), the old self-written messages that have a different user name than the current user name are placed in left. Should messages contain user ids also?

**Comment by olavim** [comment]
> Should messages contain user ids also?

It's necessary to fix that bug. But ignore the bugs for now.

**Comment by GourmetHunter** [comment]
Playing around with image editor for a minute to give you idea what my idea looks like.

(Image

**Comment by olavim** [comment]
@GourmetHunter @fergusq @pyigyli @OAarne second batch ready.

- Yes I'm aware the user listing takes a huge amount of space. I think it looks cool. This was also partly done because if the chat section takes all the screen space, it's hard to use (lot of head turning).
- Improved the input section.
- The cards-section is gone. You can access your own cards by clicking the card button next to the message input field (carousel-icon). Active rules are accessed by clicking the ballot-icon in top-right corner (under theme button, which still doesn't do anything).
- Turn is now indicated by the progress bar spinner around a user's profile pic. I don't know if we'll ever really add support for real profile pictures, but IMO a default profile pic looks much better than no profile pic.
- Top navigation bar did **not** make a comeback. Instead I made the look a bit more lively by adding some color to the user list. Which takes 40% of screen space. So there's color for you.

**Comment by fergusq** [comment]
There is so much space that I think it would be better if active cards would visible in the main view and not behind a button. I also think that the chat view should be narrower, as now the user's own messages and other messages are very far apart, this also speaks for showing active cards always. Also I think that the player's hand should be visible always (unless maybe if the screen is very small), because now the hand doesn't contain buttons and the card names themselves do not take very much space.

**Comment by fergusq** [comment]
Also a bug I noticed is that the cards that have dropdown-arguments are not accepted if the player doesn't change the selected item from the default.

**Comment by olavim** [comment]
@fergusq
> Also I think that the player's hand should be visible always

If you click the show cards -button, your cards will be visible until you click the button again. I could do something similar for the active rules -section, so that you would need to explicitly press a button to toggle it on/off.

Having them visible or hidden by default is a matter of taste I guess. I'd keep them hidden by default; this is still primarily a chat.

**Comment by fergusq** [comment]
No tällä hetkellä minun melko pienellä näytölläni 30 % tilasta on täysin käyttämättä, joten ainakin meillä on tilaa näyttää enemmän asioita. Parasta tietysti olisi jos kortit menisivät valikon taakse pienemmillä näytöillä ja olisivat esillä isommilla.

Minusta tämä on yksi keskeisimmistä ongelmista Material Designin muotoilufilosofiassa: kaikkia näyttöjä kohdellaan niin kuin ne olisivat pieniä mobiilinäyttöjä, vaikka tietokoneen syöte ja tuloste on lähtökohtaisesti hyvin erilaista ja siksi muotoilun ja asettelunkin pitäisi olla erilaista.

Meidän pitänee kysyä asiakkaalta hänen mielipidettään. Hänen mallikuvassaanhan puolet näytöstä oli varattu korteille.

**Comment by olavim** [comment]
@fergusq 
> Minusta tämä on yksi keskeisimmistä ongelmista Material Designin muotoilufilosofiassa: kaikkia näyttöjä kohdellaan niin kuin ne olisivat pieniä mobiilinäyttöjä, vaikka tietokoneen syöte ja tuloste on lähtökohtaisesti hyvin erilaista ja siksi muotoilun ja asettelunkin pitäisi olla erilaista.

Material Design kannustaa responsiiviseen muotoiluun, ja siitä huolimatta suurin osa material ui:lla tehdyistä sivustoista joilla olen vieraillut ei kyllä todellakaan ole ollut kovin mobiiliystävällisiä. Tämä on pitkälti koodaajasta kiinni, millaista haluaa tehdä.

> Parasta tietysti olisi jos kortit menisivät valikon taakse pienemmillä näytöillä ja olisivat esillä isommilla.

Kortit ja säännöt on nyt oletusarvona näkyvillä tarpeeksi suurilla näytöillä. Ne saa edelleen piiloon nappia painamalla.

File: src/client/scenes/ChatRoom.tsx [hunk 1]
```diff
@@ -126,18 +179,84 @@ const styles = (theme: Theme) => createStyles({
...
-		fontWeight: 'bold',
-		margin: '2px 0 2px 5px'
+		fontSize: '1.6rem',
+		fontWeight: 400,
+		textAlign: 'center',
+		margin: '2rem 0.4rem'
+	},
+	userListArea: {
+		flex: 0.8,
+		maxWidth: '80rem',
```
[the remark this answers is not in the document]
Comment (olavim) [reply src/client/scenes/ChatRoom.tsx]: Käyttäjälistan suuri koko on tahallista. Sen tarkoitus on samaan aikaan pienentää chatin kokoa, ja tuoda sitä keskemmälle. Omalla 2560px leveällä näytöllä esimerkiksi max `30rem` lista tekee chatista aivan liian suuren (sama asia mistä itse huomautit, kun säännöt eivät olleet viemässä tilaa oikealla puolella).

Pienemmillä leveyksillä, kuten 1440px (tyypillinen HiDPI leveys) ja 1280px käyttäjälista oli kyllä vähän liian iso. Laskin sitä varten `maxWidth`in `60rem`iin.

**Comment by olavim** [comment]
@fergusq 
>Tärkein korjattava asia on enää se, että näkymä menee rikki, jos viestejä on liian paljon ja ne menevät näytön yli

Eikä mene, korjasin tämän jossain edellisessä commitissa.

File: src/client/components/Card.tsx [hunk 2]
```diff
@@ -177,17 +125,14 @@ class OwnCard extends React.Component<OwnCardProps, OwnCardState> {
...
+			<div className={classes.cardContainer} onClick={this.handleClick}>
 				<div className={classes.cardName}>
 					{this.props.card.name}
 				</div>
-				<div className={classes.cardDescription}>
-					{this.props.card.description}
-				</div>
 				<div className={classes.playButtonContainer}>
-					{Object.keys(this.props.card.parameterTypes).map(key => {
+					{/*Object.keys(this.props.card.parameterTypes).map(key => {
```
[the remark this answers is not in the document]
Comment (olavim) [reply src/client/components/Card.tsx]: Eipä näissä muutoksissa nyt ole vielä hirveästi hyvää tyyliä muutenkaan. Anna näitä kommentteja sitten, kun teen tästä oikean PR:n.

File: src/client/components/ProfileIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+const p2 = [
+	'M26.953,0.004C12.32-0.246,0.254,11.414,0.004,26.047C-0.138,34.344,3.56,41.801,9.448,46.76',
+	'c0.385-0.336,0.798-0.644,1.257-0.894l7.907-4.313c1.037-0.566,1.683-1.653,1.683-2.835v-3.24c0,0-2.321-2.776-3.206-6.633',
+	'c-0.734-0.475-1.226-1.296-1.226-2.231v-3.546c0-0.78,0.347-1.477,0.886-1.965v-5.126c0,0-1.053-7.977,9.75-7.977',
+	's9.75,7.977,9.75,7.977v5.126c0.54,0.488,0.886,1.185,0.886,1.965v3.546c0,1.192-0.8,2.195-1.886,2.53',
+	'c-0.605,1.881-1.478,3.674-2.632,5.304c-0.291,0.411-0.563,0.759-0.801,1.03V38.8c0,1.223,0.691,2.342,1.785,2.888l8.467,4.233',
+	'c0.508,0.254,0.967,0.575,1.39,0.932c5.71-4.762,9.399-11.882,9.536-19.9C53.246,12.32,41.587,0.254,26.953,0.004z'
+];
+
+const ProfileIcon = (props: any) => (
```
[the remark this answers is not in the document]
Comment (olavim) [reply src/client/components/ProfileIcon.tsx]: Proof of concept. Kuva muuttuu vielä varsinaiseen PR:ään tästä.

**Comment by fergusq** [comment]
Viestit, jotka menevät näytön yli, eivät toimi ainakaan Firefoxissa.

kuva virheestä

**Comment by olavim** [comment]
> Viestit, jotka menevät näytön yli, eivät toimi ainakaan Firefoxissa.

Katsoin kyllä mielestäni että toimii myös firefoxissa :S ehkä mennyt rikki myöhemmässä commitissa. Korjaan.

**Comment by olavim** [comment]
@fergusq @GourmetHunter @OAarne @pyigyli PR is now ready for code review. Feel free to still voice your opinions if you don't agree with some design choices. There's obviously a lot of changed code due to extensive rewrite of styles, but hopefully that won't intimidate you too much.

**Comment by GourmetHunter** [comment]
All in all, this looks really good, visually. I will take a closer look at the code later and comment on it.

### Code / Styling

I think we should use react-custom-scrollbars instead of standard scrollbars.
This is really simple actually,
```
yarn add react-custom-scrollbars
```

Then import
```
import { Scrollbars } from 'react-custom-scrollbars';
```

Then wrap part you want to have a scrollbar in
```
<Scrollbars>
    ...
</Scrollbars>
```

I had some problems getting this working on horizontal bars, but it works just fine vertical ones.

### Functionality

One issue (which I do find acceptable trade off because of the increase of, at least, visual guality of the program) is that the join / create room page is hideously ugly now, even the old one was better, and lacks functionality attached to the master branch (such as changing theme and language).

What should be corrected on functionality side before pushing this to master is to add Language change button (and function) from master branch. (Avoiding downgrading on functional side.)

### Visual

New color themes are excellent, especially the dark theme. And I am very happy that the size of the right section is now *static*. The size of the section however is still not pleasing to me, however I think there's no need to change it, and instead I can create a theme later that changes the size of it.

**Comment by olavim** [comment]
@GourmetHunter 
> I think we should use react-custom-scrollbars instead of standard scrollbars

Sure, why not. Since functionality on mobile is not an issue, it's probably a good idea.

> the join / create room page is hideously ugly now

I completely forgot about the main menu 😄 I'll get it in order.

> add Language change button (and function) from master branch

👍

**Comment by GourmetHunter** [comment]
Nevermind the languages change functionality I mentioned, it hasn't been pushed to master yet.

**Comment by olavim** [comment]
@GourmetHunter Addressed your requests in the latest commit.
- Styles for main menu
- Small theme changes
- Custom scrollbars

File: src/client/components/App.tsx [hunk 4]
```diff
@@ -110,19 +110,14 @@ class App extends React.Component<Props & RouteComponentProps & WithStyles<typeo
...
-					this.setState({ownCards: []});
-					break;
 				case 'ROOM_STATE':
 					this.setState({
 						users: msg.users,
 						userMap: msg.users.reduce((m, u) => ({...m, [u.id]: u}), {}),
 						activeCards: msg.enabledRules,
 						turnUserId: msg.turnUserId,
-						nickname: msg.nickname
+						user: msg.users.find(u => u.nickname === msg.nickname) || null,
```
[the remark this answers is not in the document]
Comment (olavim) [reply src/client/components/App.tsx]: Hyvä kun huomasit, kyllä pitäisi.

**Pull request merged**